sof: update to latest SOF master

nashif opened the pull request:
Sync with SOF tree 385cf89c7

Fixes #31301

nashif committed:
sof: update to latest SOF master

Sync with SOF tree 385cf89c7

Fixes #31301

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

lyakh commented on west.yml:
```diff
@@ -118,7 +118,7 @@ manifest:
       path: modules/debug/segger
     - name: sof
-      revision: 9e75cce8bc1c649672931a4d772b2411a890804b
+      revision: pull/6/head
```
not sure this is more intuitive than just a SHA1, but apparently it works

nashif replied:
this is just pointing to an open PR, once the PR is merged, it will need to move to a SHA.

nashif committed:
boards: cavs15: fixed link to private key

Fixed link to the private key. Point to new location.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
west: sign: add new option -D for configurations

rimage signing tool requires the path to the configuration directory.
This can be fetched using cmake cache if we are building with a module
that has rimage integrated like the sof module, however, we should be
able to sign images that were built standalone and without SOF, so
support this new option to provide the location of the configuration
files for the tool.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
west: rimage: get config dir from command line

If we do not have the configuration path via cmake, then check if the
configuration path was provided on the command line and use it instead,
if both fail, bail out and die.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
